kong/core/router.lua

```diff
@@ -831,7 +831,7 @@ function _M.new(apis)
       end
     end
 
-    local req_host = ngx.var.http_host
+    local req_host = ngx.var.http_host or ""
 
```

Comment:
Should we allow this bevahior only when the request protocol is HTTP/1.0? The `Host` header is required for 1.1 requests; i would proffer that Kong should require 1.1 clients to follow the spec and not accept invalid requests.

Reply:
Yes, the goal of this patch is to support this behavior only for HTTP/1.0 (as discussed with @rainest). There is probably nothing else to add in the router since NGINX should disallow HTTP/1.1 requests without the Host head already (hence why the ask for the test so we avoid a future regression). We could also add a check in the router but it seems unnecessary to me...